gc/base/Configuration.cpp

```diff
@@ -525,3 +525,27 @@ MM_Configuration::createParallelDispatcher(MM_EnvironmentBase *env, omrsig_handl
 {
 	return MM_ParallelDispatcher::newInstance(env, handler, handler_arg, defaultOSStackSize);
 }
+
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+bool
+MM_Configuration::reinitializeForRestore(MM_EnvironmentBase* env)
+{
+	MM_GCExtensionsBase *extensions = env->getExtensions();
+
+	/* The initialization of the GC thread count can only happen if it is not enforced by the user. */
+	initializeGCThreadCount(env);
+
+	/* Currently, threads don't shutdown during reinitialization, so it is important to
+	 * prevent the thread count from dropping below the number of checkpoint threads.
+	 * This adjustment may no longer be necessary in the future once the shutdown behavior
+	 * of dispatcher threads during restore is thoroughly tested.
+	 */
+	extensions->gcThreadCount = OMR_MAX(
```

Comment:
temporarily couple of those lines that are migrated from J9 will be executed twice, right? hopefully, not causing major problems

Reply:
Yup, `initializeGCThreadCount` and MAX adjustment will be just done twice, no issue with that though.